### PR TITLE
JAVA-29472 Upgrade open-api-generator-maven-plugin to latest version

### PR DESCRIPTION
**Pull request by anuragkumawat**

### CODE REVIEW

File: spring-swagger-codegen/custom-validations-opeanpi-codegen/src/main/resources/openapi/templates/beanValidationCore.mustache [hunk 1]
```diff
@@ -1,26 +1,25 @@
...
 minLength not set, maxLength set
-}}{{^minLength}}{{#maxLength}}@Size(max={{maxLength}}) {{/maxLength}}{{/minLength}}{{!
+}}{{^minLength}}{{#maxLength}}@Size(max = {{.}}) {{/maxLength}}{{/minLength}}{{!
 @Size: minItems && maxItems set
-}}{{#minItems}}{{#maxItems}}@Size(min={{minItems}},max={{maxItems}}) {{/maxItems}}{{/minItems}}{{!
+}}{{#minItems}}{{#maxItems}}@Size(min = {{minItems}}, max = {{maxItems}}) {{/maxItems}}{{/minItems}}{{!
 @Size: minItems set, maxItems not
-}}{{#minItems}}{{^maxItems}}@Size(min={{minItems}}) {{/maxItems}}{{/minItems}}{{!
+}}{{#minItems}}{{^maxItems}}@Size(min = {{minItems}}) {{/maxItems}}{{/minItems}}{{!
 @Size: minItems not set && maxItems set
```
Comment (anastasiosioannidis) [inline spring-swagger-codegen/custom-validations-opeanpi-codegen/src/main/resources/openapi/templates/beanValidationCore.mustache:13]: Hi @anuragkumawat 

When you have time, could you point me to the documentation that helped you do the below "migration" of this specific part (from the 4 red lines 15-19, to the next 4 green lines)

File: spring-swagger-codegen/custom-validations-opeanpi-codegen/src/main/resources/openapi/templates/enumOuterClass.mustache
Comment (anastasiosioannidis) [inline spring-swagger-codegen/custom-validations-opeanpi-codegen/src/main/resources/openapi/templates/enumOuterClass.mustache:1]: Hi @anuragkumawat 

Can you summarize how you resulted in creating this new enumOuterClass.mustache?
Just trying to understand, thanks.

File: spring-swagger-codegen/custom-validations-opeanpi-codegen/src/main/resources/openapi/templates/beanValidationCore.mustache (hunk 1 above)
Comment (anuragkumawat) [reply spring-swagger-codegen/custom-validations-opeanpi-codegen/src/main/resources/openapi/templates/beanValidationCore.mustache:13]: The link I used is https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/JavaSpring/beanValidationCore.mustache

Thanks
